common/namespace/cache.go

```diff
@@ -399,15 +401,28 @@ func (c *namespaceCache) GetNamespaceName(
 	return entry.info.Name, nil
 }
 
+func (c *namespaceCache) Refresh() {
+	replyCh := make(chan struct{})
+	c.triggerRefreshCh <- replyCh
+	<-replyCh
+}
+
 func (c *namespaceCache) refreshLoop() {
 	timer := time.NewTicker(CacheRefreshInterval)
 	defer timer.Stop()
 
+	// Put timer events on our channel so we can select on just one below.
+	go func() {
+		for range timer.C {
+			c.triggerRefreshCh <- nil
```

Comment:
Maybe non-blocking here? If `triggerRefreshCh` is full then we know that a refresh is about to occur

Reply:
That's true. I was thinking for tests it didn't really matter, but non-blocking makes more sense in general. I'll do that.